mls-rs/src/group/proposal_filter/filtering.rs

```diff
@@ -106,6 +113,13 @@ where
         let proposals = filter_out_reinit_if_other_proposals(strategy.is_ignore(), proposals)?;
         let proposals = filter_out_external_init(strategy, proposals)?;
 
+        #[cfg(all(
+            feature = "by_ref_proposal",
+            feature = "custom_proposal",
+            feature = "self_remove_proposal"
+        ))]
+        check_for_remove_and_self_remove_same_leaf(&proposals)?;
```

Comment:
This is where it should filter out proposal or error out according to `strategy`. See `filter_out_external_init` above. BTW a commit with remove and self remove would fail on `batch_edit` anyway - `blank_leaf_node` function I think fails if it's called on a leaf twice.

Reply:
Ah that makes sense. Done!